pytorch_lightning/callbacks/fault_tolerance.py

```diff
@@ -0,0 +1,49 @@
+# Copyright The PyTorch Lightning team.
```

Comment:
I tried to put this in `pytorch_lightning/utilities/auto_restart.py` initially but couldn't resolve the circular dependencies.

Reply:
Can we make the file private, like pytorch_lightning/callbacks/_fault_tolerance.py